apluslms_roman/backends/docker.py

```diff
@@ -99,20 +103,41 @@ def _run_opts(self, task, step):
 
     def prepare(self, task, observer):
         client = self._client
+        images = self._cache.images
+        day = timedelta(days=1)
         for step in task.steps:
             observer.step_preflight(step)
             image, tag = step.img.split(':', 1)
+
+            last_updated = images.get(step.img, None)
+            last_updated = (day if not last_updated
+                else datetime.now() - last_updated)
+
             try:
-                img = client.images.get(step.img)
+                client.images.get(step.img)
+                img_found = True
             except docker.errors.ImageNotFound:
+                img_found = False
+
+            if not img_found or last_updated >= day:
```

Comment:
should there be a way to force docker to download the image? (like if the user knows there's an update and wants to download it) besides deleting the image from the system or editing the cache...

Reply:
`roman prepare -f` or something would probably be good

Reply:
okay I'll add that to the `prepare_only` branch